--- src/TerrariaNetCore/Terraria/Testing/DebugOptions.cs
+++ src/Unified/Terraria/Testing/DebugOptions.cs
@@ -1,11 +_,11 @@
-using Microsoft.Xna.Framework;
+using System.Diagnostics;
 using Terraria.Localization;
 
 namespace Terraria.Testing;
 
 public static class DebugOptions
 {
-	public static bool enableDebugCommands = false;
+	public static bool enableDebugCommands = Debugger.IsAttached;
 	public static bool Shared_ReportCommandUsage = true;
 	public static bool Shared_RandomizeProjectileSlots = false;
 	public static int Shared_ServerPing = 0;
