--- src/TerrariaNetCore/Terraria/UI/ChestUI.cs
+++ src/Unified/Terraria/UI/ChestUI.cs
@@ -39,18 +_,19 @@
 
 	public static void UpdateHover(int ID, bool hovering)
 	{
+		float scaling = 0.05f * Main.DrawDeltaTime;
 		if (hovering) {
 			if (!ButtonHovered[ID])
 				SoundEngine.PlaySound(12);
 
 			ButtonHovered[ID] = true;
-			ButtonScale[ID] += 0.05f;
+			ButtonScale[ID] += scaling;
 			if (ButtonScale[ID] > 1f)
 				ButtonScale[ID] = 1f;
 		}
 		else {
 			ButtonHovered[ID] = false;
-			ButtonScale[ID] -= 0.05f;
+			ButtonScale[ID] -= scaling;
 			if (ButtonScale[ID] < 0.75f)
 				ButtonScale[ID] = 0.75f;
 		}
@@ -119,7 +_,10 @@
 		Vector2 vector = new Vector2(504f, Main.instance.invBottom);
 		ChatManager.DrawColorCodedStringWithShadow(spritebatch, value, text, vector, color, 0f, Vector2.Zero, Vector2.One, -1f, 1.5f);
 		if (Main.editChest) {
+			/*
 			vector.X += value.MeasureString(text).X;
+			*/
+			vector.X += ChatManager.GetStringSize(value, text, Vector2.One).X;
 			Main.instance.SetIMEPanelAnchor(vector + new Vector2(0f, 56f), 0f);
 			string compositionString = Platform.Get<IImeService>().CompositionString;
 			if (compositionString != null && compositionString.Length > 0) {
@@ -127,7 +_,11 @@
 				vector.X += value.MeasureString(compositionString).X;
 			}
 
+			/*
 			if (++Main.instance.textBlinkerCount >= 20) {
+			*/
+			Main.instance.textBlinkerCount += 1 * Main.DrawDeltaTime;
+			if (Main.instance.textBlinkerCount >= 20) {
 				Main.instance.textBlinkerState = ((Main.instance.textBlinkerState == 0) ? 1 : 0);
 				Main.instance.textBlinkerCount = 0;
 			}
