--- src/TerrariaNetCore/Terraria/Localization/LanguageManager.cs
+++ src/Unified/Terraria/Localization/LanguageManager.cs
@@ -189,7 +_,7 @@
 					IAssetRepository assetRepository = Main.instance.Services.Get<IAssetRepository>();
 					if (assetRepository != null && assetRepository.AssetLoadFailHandler != null) {
 						string assetName = item + extension;
-						assetRepository.AssetLoadFailHandler(assetName, AssetLoadException.FromAssetException(assetName, innerException));
+						assetRepository.AssetLoadFailHandler(assetName, new IOException(assetName, innerException));
 					}
 				}
 			}
