--- src/TerrariaNetCore/ReLogic/Content/Sources/XnaDirectContentSource.cs
+++ src/Unified/ReLogic/Content/Sources/XnaDirectContentSource.cs
@@ -1,9 +_,64 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using ReLogic.Utilities;
 
 namespace ReLogic.Content.Sources;
 
+public class XnaDirectContentSource : ContentSource
+{
+	private readonly string[] _rootDirectories;
+
+	private readonly Dictionary<string, CaseInsensitiveFileLookup> _fileLookups = new(StringComparer.OrdinalIgnoreCase);
+
+	public override string FileWatcherPath => null;
+
+	public XnaDirectContentSource(IEnumerable<string> rootDirectories)
+	{
+		_rootDirectories = rootDirectories.Select(AssetPathHelper.CleanPath).ToArray();
+		foreach (string rootDir in _rootDirectories)
+		{
+			var lookup = new CaseInsensitiveFileLookup(
+				rootDir,
+				CaseInsensitiveFileLookup.LookupKeyMode.RelativePath,
+				true);
+
+			_fileLookups.TryAdd(rootDir, lookup);
+		}
+		Refresh();
+	}
+
+	public override Stream OpenStream(string assetName)
+	{
+		try {
+			//return File.OpenRead(_rootDirectories.Select(rootDir => Path.Combine(rootDir, assetName)).First(File.Exists));
+			foreach (var (_, fileLookup) in _fileLookups)
+			{
+				if (!fileLookup.TryGetFile(assetName, out string assetPath))
+					continue;
+
+				return File.OpenRead(assetPath);
+			}
+
+			throw new FileNotFoundException(assetName);
+		}
+		catch (Exception innerException) {
+			throw AssetLoadException.FromMissingAsset(assetName, innerException);
+		}
+	}
+
+	public override void Refresh()
+	{
+		SetAssetNames(
+			_rootDirectories
+				.SelectMany(rootDir => Directory.GetFiles(rootDir, "*.xnb", SearchOption.AllDirectories).Select(path => path.Substring(rootDir.Length + 1)))
+				.ToHashSet()
+		);
+	}
+}
+
+/*
 public class XnaDirectContentSource : IContentSource
 {
 	private readonly string _rootDirectory;
@@ -75,3 +_,4 @@
 
 	public bool TryGetRejections(List<string> rejectionReasons) => _rejections.TryGetRejections(rejectionReasons);
 }
+*/
