--- src/TerrariaNetCore/Terraria/Cloud.cs
+++ src/Unified/Terraria/Cloud.cs
@@ -1,15 +_,16 @@
 using Microsoft.Xna.Framework;
 using Microsoft.Xna.Framework.Graphics;
 using Terraria.GameContent;
+using Terraria.Unified.Time;
 using Terraria.Utilities;
 
 namespace Terraria;
 
 public class Cloud
 {
-	public Vector2 position;
+	public DeltaVector2<CurrentDelta> position;
 	public float scale;
-	public float rotation;
+	public DeltaFloat<CurrentDelta> rotation;
 	public float rSpeed;
 	public float sSpeed;
 	public bool active;
@@ -17,7 +_,7 @@
 	public int type;
 	public int width;
 	public int height;
-	public float Alpha;
+	public DeltaFloat<CurrentDelta> Alpha;
 	public bool kill;
 	private static UnifiedRandom rand = new UnifiedRandom();
 	public static Vector2? lastCameraCenter;
