--- src/TerrariaNetCore/Terraria/MessageBuffer.cs
+++ src/Unified/Terraria/MessageBuffer.cs
@@ -23,6 +_,7 @@
 using Terraria.Social.Steam;
 using Terraria.Testing;
 using Terraria.UI;
+using Terraria.Unified.Features;
 
 namespace Terraria;
 
@@ -963,8 +_,10 @@
 				if (!WorldGen.InWorld(num132, num133, 3))
 					break;
 
+				/*
 				if (Main.tile[num132, num133] == null)
 					Main.tile[num132, num133] = new Tile();
+				*/
 
 				if (Main.netMode == 2) {
 					if (!flag11) {
@@ -1149,11 +_,13 @@
 				BitsByte bitsByte22 = (byte)0;
 				BitsByte bitsByte23 = (byte)0;
 				BitsByte bitsByte24 = (byte)0;
-				Tile tile4 = null;
+				Tile tile4 = default;
 				for (int num90 = num86; num90 < num86 + num88; num90++) {
 					for (int num91 = num87; num91 < num87 + num89; num91++) {
+						/*
 						if (Main.tile[num90, num91] == null)
 							Main.tile[num90, num91] = new Tile();
+						*/
 
 						tile4 = Main.tile[num90, num91];
 						bool flag7 = tile4.active();
@@ -1633,10 +_,19 @@
 				int num24 = reader.ReadByte();
 				string name = string.Empty;
 				if (num24 != 0) {
+					/*
 					if (num24 <= 20)
 						name = reader.ReadString();
 					else if (num24 != 255)
 						num24 = 0;
+					*/
+
+					if (num24 <= ChestNameLengthExtender.PreferredMaxChestNameLength) {
+						name = reader.ReadString();
+					}
+					else if (num24 != 255) {
+						num24 = 0;
+					}
 				}
 
 				if (Main.netMode == 1) {
@@ -2044,10 +_,12 @@
 						Netplay.Clients[whoAmI].SpamWater += 1f;
 				}
 
+				/*
 				if (Main.tile[num10, num11] == null)
 					Main.tile[num10, num11] = new Tile();
+				*/
 
-				lock (Main.tile[num10, num11]) {
+				// lock (Main.tile[num10, num11]) {
 					Main.tile[num10, num11].liquid = b2;
 					Main.tile[num10, num11].liquidType(liquidType);
 					if (Main.netMode == 2) {
@@ -2057,7 +_,7 @@
 					}
 
 					break;
-				}
+				// }
 			}
 			case 49:
 				if (Netplay.Connection.State == 6) {
