--- src/TerrariaNetCore/Terraria/Mount.cs
+++ src/Unified/Terraria/Mount.cs
@@ -4909,7 +_,7 @@
 		}
 
 		int type = _type;
-		if (type == 50 && texture2D != null) {
+		if (type == 50 && texture2D != null && texture2D != Asset<Texture2D>.DefaultValue) {
 			PlayerQueenSlimeMountTextureContent queenSlimeMount = TextureAssets.RenderTargets.QueenSlimeMount;
 			queenSlimeMount.Request();
 			if (queenSlimeMount.IsReady)
