--- src/TerrariaNetCore/Terraria/NetMessage.cs
+++ src/Unified/Terraria/NetMessage.cs
@@ -14,6 +_,7 @@
 using Terraria.Map;
 using Terraria.Social;
 using Terraria.Testing;
+using Terraria.Unified.Features;
 
 namespace Terraria;
 
@@ -794,11 +_,30 @@
 
 					if (number2 == 1f) {
 						string text3 = text.ToString();
+
+						/*
 						num10 = (byte)text3.Length;
 						if (num10 == 0 || num10 > 20)
 							num10 = 255;
 						else
 							text2 = text3;
+						*/
+
+						num10 = (byte)Math.Clamp(text3.Length, 0, 255);
+						if (num10 == 0 || num10 > ChestNameLengthExtender.VanillaMaxChestNameLength) {
+							if (num10 > ChestNameLengthExtender.ModdedMaxChestNameLength) {
+								num10 = byte.MaxValue;
+							}
+							else if (ChestNameLengthExtender.BypassServerNameLengthChecks) {
+								// Clamp the actual length if we want to bypass
+								// name checks, since the length is useless.
+								num10 = ChestNameLengthExtender.VanillaMaxChestNameLength;
+							}
+						}
+
+						if (num10 != byte.MaxValue) {
+							text2 = text3;
+						}
 					}
 
 					writer.Write((short)number);
@@ -1723,11 +_,11 @@
 		int num6 = 0;
 		byte b = 0;
 		byte[] array = new byte[16];
-		Tile tile = null;
+		Tile? tile = null;
 		for (int i = yStart; i < yStart + height; i++) {
 			for (int j = xStart; j < xStart + width; j++) {
 				Tile tile2 = Main.tile[j, i];
-				if (tile2.isTheSameAs(tile) && TileID.Sets.AllowsSaveCompressionBatching[tile2.type]) {
+				if (tile.HasValue && tile2.isTheSameAs(tile.Value) && TileID.Sets.AllowsSaveCompressionBatching[tile2.type]) {
 					num4 = (short)(num4 + 1);
 					continue;
 				}
@@ -2005,16 +_,19 @@
 
 	public static void DecompressTileBlock_Inner(BinaryReader reader, int xStart, int yStart, int width, int height)
 	{
-		Tile tile = null;
+		Tile tile = default;
 		int num = 0;
 		for (int i = yStart; i < yStart + height; i++) {
 			for (int j = xStart; j < xStart + width; j++) {
 				if (num != 0) {
 					num--;
+					/*
 					if (Main.tile[j, i] == null)
 						Main.tile[j, i] = new Tile(tile);
 					else
 						Main.tile[j, i].CopyFrom(tile);
+					*/
+					Main.tile[j, i].CopyFrom(tile);
 
 					continue;
 				}
