--- src/TerrariaNetCore/Terraria/Player.cs
+++ src/Unified/Terraria/Player.cs
@@ -40,7 +_,7 @@
 
 namespace Terraria;
 
-public class Player : Entity, IFixLoadedData
+public partial class Player : Entity, IFixLoadedData
 {
 	public static class BuilderAccToggleIDs
 	{
@@ -17682,7 +_,7 @@
 			Hurt(PlayerDeathReason.LegacyDefault(), attackDamage_ScaledByDifficulty, Main.npc[Main.wofNPCIndex].direction);
 		}
 
-		if (!gross && position.Y > (float)((Main.maxTilesY - 250) * 16) && position.X > num - (float)Main.MaxWorldViewSize.X && position.X < num + (float)Main.MaxWorldViewSize.X) {
+		if (!gross && position.Y > (float)((Main.maxTilesY - 250) * 16) && position.X > num - (float)Main.MaxGameplayViewSize.X && position.X < num + (float)Main.MaxGameplayViewSize.X) {
 			AddBuff(37, 10);
 			SoundEngine.PlaySound(4, (int)Main.npc[Main.wofNPCIndex].position.X, (int)Main.npc[Main.wofNPCIndex].position.Y, 10);
 		}
@@ -18882,7 +_,7 @@
 
 	public static Tile GetFloorTile(int x, int y)
 	{
-		Tile result = null;
+		Tile result = default;
 		if (Main.tile[x - 1, y] == null)
 			Main.tile[x - 1, y] = new Tile();
 
@@ -18902,7 +_,11 @@
 		return result;
 	}
 
+	/*
 	public static int GetFloorTileType(int x, int y) => ((int?)GetFloorTile(x, y)?.type) ?? (-1);
+	*/
+
+	public static int GetFloorTileType(int x, int y) => GetFloorTile(x, y).type;
 
 	private void MakeFloorDust(bool Falling, int type, int frameX, int frameY, int paintColor)
 	{
@@ -19490,6 +_,8 @@
 		if (whoAmI != Main.myPlayer) {
 			int num2 = (int)(position.X + (float)(width / 2)) / 16;
 			int num3 = (int)(position.Y + (float)(height / 2)) / 16;
+
+			/*
 			if (!WorldGen.InWorld(num2, num3, 4))
 				flag = true;
 			else if (Main.tile[num2, num3] == null)
@@ -19502,6 +_,10 @@
 				flag = true;
 			else if (Main.tile[num2, num3 + 3] == null)
 				flag = true;
+			*/
+
+			if (Main.netMode == NetmodeID.MultiplayerClient && !Main.sectionManager.TilesLoaded(num2 - 3, num3 - 3, num2 + 3, num3 + 3))
+				flag = true;
 
 			if (flag) {
 				outOfRange = true;
@@ -20103,6 +_,7 @@
 
 			TrySyncingInput();
 			UpdateNearbyCraftingTiles();
+			UpdateCraftingReachAnimFrames();
 			HandleBeingInChestRange();
 			tileEntityAnchor.GetTileEntity()?.OnPlayerUpdate(this);
 		}
@@ -28746,6 +_,8 @@
 
 	public void AdjTiles()
 	{
+		TilesWithinCraftingReach.Clear();
+
 		Array.Clear(adjTile, 0, adjTile.Length);
 		oldAdjWaterSource = adjWaterSource;
 		adjWaterSource = false;
@@ -28756,13 +_,85 @@
 		alchemyTable = false;
 		Rectangle tileRegion = TileReachCheckSettings.Simple.GetTileRegion(this, ateArtisanBread ? 4 : 0);
 		tileRegion = WorldUtils.ClampToWorld(tileRegion);
+
+		CraftingReachRectangle = tileRegion;
+
 		for (int i = tileRegion.Left; i <= tileRegion.Right; i++) {
 			for (int j = tileRegion.Top; j <= tileRegion.Bottom; j++) {
 				Tile tile = Main.tile[i, j];
 				if (tile.active()) {
 					SetAdjTile(tile.type);
+					/*
 					if (TileID.Sets.CountsAsWaterForCrafting[tile.type])
 						adjWaterSource = true;
+					*/
+
+					var isACraftingStation = Recipe.TileUsedInRecipes[tile.type];
+					if (TileID.Sets.CountsAsWaterForCrafting[tile.type]) {
+						adjWaterSource = true;
+						isACraftingStation = true;
+					}
+
+					if (isACraftingStation) {
+						TilesWithinCraftingReach.Add(new Point16(i, j));
+
+						if (i == tileRegion.Left) {
+							for (int x = tileRegion.Left - 1; x >= tileRegion.Left - 3; x--) {
+								if (x < 0) {
+									break;
+								}
+
+								if (tile.type == Main.tile[x, j].type) {
+									TilesWithinCraftingReach.Add(new Point16(x, j));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (i == tileRegion.Right) {
+							for (int x = tileRegion.Right + 1; x <= tileRegion.Right + 3; x++) {
+								if (x >= Main.maxTilesX) {
+									break;
+								}
+
+								if (tile.type == Main.tile[x, j].type) {
+									TilesWithinCraftingReach.Add(new Point16(x, j));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (j == tileRegion.Top) {
+							for (int y = tileRegion.Top - 1; y >= tileRegion.Top - 3; y--) {
+								if (y < 0) {
+									break;
+								}
+
+								if (tile.type == Main.tile[i, y].type) {
+									TilesWithinCraftingReach.Add(new Point16(i, y));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (j == tileRegion.Bottom) {
+							for (int y = tileRegion.Bottom + 1; y <= tileRegion.Bottom + 3; y++) {
+								if (y >= Main.maxTilesY) {
+									break;
+								}
+
+								if (tile.type == Main.tile[i, y].type) {
+									TilesWithinCraftingReach.Add(new Point16(i, y));
+								}
+								else {
+									break;
+								}
+							}
+						}
+					}
 				}
 
 				if (tile.liquid > 200 && tile.liquidType() == 0)
@@ -33897,8 +_,8 @@
 	public void HorsemansBlade_SpawnPumpkin(int npcIndex, int dmg, float kb)
 	{
 		Vector2 center = Main.npc[npcIndex].Center;
-		int y = Main.MaxWorldViewSize.Y;
+		int y = Main.MaxGameplayViewSize.Y;
-		int x = Main.MaxWorldViewSize.X;
+		int x = Main.MaxGameplayViewSize.X;
 		int num = Main.rand.Next(100, 300);
 		int num2 = Main.rand.Next(100, 300);
 		num = ((Main.rand.Next(2) != 0) ? (num + (x / 2 - num)) : (num - (x / 2 + num)));
@@ -36709,7 +_,7 @@
 
 	public void LimitPointToPlayerReachableArea(ref Vector2 pointPosition)
 	{
-		Rectangle worldRect = Utils.CenteredRectangle(base.Center, Main.MaxWorldViewSize.ToVector2());
+		Rectangle worldRect = Utils.CenteredRectangle(base.Center, Main.MaxGameplayViewSize.ToVector2());
 		worldRect = WorldUtils.ClampToWorldBorders(worldRect);
 		Vector2 vector = worldRect.Center.ToVector2();
 		Vector2 vector2 = pointPosition - vector;
