--- src/TerrariaNetCore/Terraria/Player.cs
+++ src/Unified/Terraria/Player.cs
@@ -42,7 +_,7 @@
 
 namespace Terraria;
 
-public class Player : Entity, IFixLoadedData
+public partial class Player : Entity, IFixLoadedData
 {
 	public static class BuilderAccToggleIDs
 	{
@@ -15871,11 +_,14 @@
 			return;
 
 		float num3 = position.Y;
+
+		/*
 		if (Main.tile[num, num2 - 1] == null)
 			Main.tile[num, num2 - 1] = new Tile();
 
 		if (Main.tile[num, num2 + 1] == null)
 			Main.tile[num, num2 + 1] = new Tile();
+		*/
 
 		if ((!Main.tile[num, num2 - 1].active() || !Main.tileRope[Main.tile[num, num2 - 1].type]) && (!Main.tile[num, num2 + 1].active() || !Main.tileRope[Main.tile[num, num2 + 1].type]))
 			num3 = num2 * 16 + 22;
@@ -18844,8 +_,10 @@
 
 				int num3 = (int)((position.Y + (float)num2 - 44f * gravDir) / 16f);
 				if (Main.tile[num, num3] != null && Main.tile[num, num3].liquid < 128) {
+					/*
 					if (Main.tile[num, num3] == null)
 						Main.tile[num, num3] = new Tile();
+					*/
 
 					if (!Main.tile[num, num3].active() || !Main.tileSolid[Main.tile[num, num3].type] || Main.tileSolidTop[Main.tile[num, num3].type])
 						flag = false;
@@ -19271,10 +_,17 @@
 		if (gravDir == -1f)
 			num2 = (int)(position.Y - 0.1f) / 16;
 
+		/*
 		Tile floorTile = GetFloorTile(num, num2);
+		*/
+		Tile? floorTile = GetFloorTile(num, num2);
 		int num3 = -1;
+		/*
 		if (floorTile != null)
 			num3 = floorTile.type;
+		*/
+		if (floorTile.HasValue)
+			num3 = floorTile.Value.type;
 
 		if (num3 <= -1) {
 			ResetFloorFlags();
@@ -19293,7 +_,10 @@
 			num3 = -1;
 
 		if (!wet && !mount.Cart)
+			/*
 			MakeFloorDust(Falling, num3, floorTile.frameX, floorTile.frameY, floorTile.color());
+			*/
+			MakeFloorDust(Falling, num3, floorTile.Value.frameX, floorTile.Value.frameY, floorTile.Value.color());
 	}
 
 	public void ResetFloorFlags()
@@ -19305,9 +_,14 @@
 		runningOnSand = false;
 	}
 
+	/*
 	public static Tile GetFloorTile(int x, int y)
+	*/
+	public static Tile? GetFloorTile(int x, int y)
 	{
-		Tile result = null;
+		Tile? result = null;
+
+		/*
 		if (Main.tile[x - 1, y] == null)
 			Main.tile[x - 1, y] = new Tile();
 
@@ -19316,6 +_,7 @@
 
 		if (Main.tile[x, y] == null)
 			Main.tile[x, y] = new Tile();
+		*/
 
 		if (Main.tile[x, y].nactive() && Main.tileSolid[Main.tile[x, y].type])
 			result = Main.tile[x, y];
@@ -19554,9 +_,16 @@
 		if (position.Y < Main.topWorld + (float)num) {
 			Main.cameraY = 0f;
 			if (Main.remixWorld || forcedGravity > 0) {
+				// TU(#28): Restrict border movement killing the player to the
+				// owning client.
+				/*
 				if (position.Y < Main.topWorld + (float)num - (float)height && !dead) {
+				*/
+				if (position.Y < Main.topWorld + (float)num - (float)height && !dead && whoAmI == Main.myPlayer) {
 					KillMe(PlayerDeathReason.ByOther(19), 10.0, 0);
+					/*
 					if (whoAmI == Main.myPlayer)
+					*/
 						AchievementsHelper.NotifyProgressionEvent(44);
 				}
 
@@ -19582,7 +_,12 @@
 
 		bool skyblockWorld = Main.skyblockWorld;
 		if (position.Y > Main.bottomWorld - (float)num) {
+			// TU(#28): Restrict border movement killing the player to the
+			// owning client.
+			/*
 			if (skyblockWorld && !dead)
+			*/
+			if (skyblockWorld && !dead && whoAmI == Main.myPlayer)
 				KillMe(PlayerDeathReason.ByOther(21), 10.0, 0);
 
 			Main.cameraY = 0f;
@@ -19916,6 +_,8 @@
 		if (whoAmI != Main.myPlayer) {
 			int num2 = (int)(position.X + (float)(width / 2)) / 16;
 			int num3 = (int)(position.Y + (float)(height / 2)) / 16;
+
+			/*
 			if (!WorldGen.InWorld(num2, num3, 4))
 				flag = true;
 			else if (Main.tile[num2, num3] == null)
@@ -19928,6 +_,10 @@
 				flag = true;
 			else if (Main.tile[num2, num3 + 3] == null)
 				flag = true;
+			*/
+
+			if (Main.netMode == NetmodeID.MultiplayerClient && !Main.sectionManager.TilesLoaded(num2 - 3, num3 - 3, num2 + 3, num3 + 3))
+				flag = true;
 
 			if (flag) {
 				outOfRange = true;
@@ -20551,6 +_,7 @@
 
 			TrySyncingInput();
 			UpdateNearbyCraftingTiles();
+			UpdateCraftingReachAnimFrames();
 			HandleBeingInChestRange();
 			tileEntityAnchor.GetTileEntity()?.OnPlayerUpdate(this);
 		}
@@ -20649,6 +_,7 @@
 		if (tileTargetY < 5)
 			tileTargetY = 5;
 
+		/*
 		if (Main.tile[tileTargetX - 1, tileTargetY] == null)
 			Main.tile[tileTargetX - 1, tileTargetY] = new Tile();
 
@@ -20657,6 +_,7 @@
 
 		if (Main.tile[tileTargetX, tileTargetY] == null)
 			Main.tile[tileTargetX, tileTargetY] = new Tile();
+		*/
 
 		if (inventory[selectedItem].axe > 0 && !Main.tile[tileTargetX, tileTargetY].active() && inventory[selectedItem].createWall <= 0 && (inventory[selectedItem].hammer <= 0 || inventory[selectedItem].axe != 0)) {
 			if (Main.tile[tileTargetX - 1, tileTargetY].active() && Main.tile[tileTargetX - 1, tileTargetY].type == 323) {
@@ -21295,11 +_,14 @@
 					int num35 = num34 * 16 + 8 - width / 2;
 					float y2 = position.Y;
 					y2 = num32 * 16 + 22;
+
+					/*
 					if (Main.tile[num34, num32 - 1] == null)
 						Main.tile[num34, num32 - 1] = new Tile();
 
 					if (Main.tile[num34, num32 + 1] == null)
 						Main.tile[num34, num32 + 1] = new Tile();
+					*/
 
 					if (WorldGen.IsRope(num34, num32 - 1) || WorldGen.IsRope(num34, num32 + 1))
 						y2 = num32 * 16 + 22;
@@ -21332,8 +_,10 @@
 			if (velocity.X != 0f)
 				pulley = false;
 
+			/*
 			if (Main.tile[num31, num32] == null)
 				Main.tile[num31, num32] = new Tile();
+			*/
 
 			if (!WorldGen.IsRope(num31, num32))
 				pulley = false;
@@ -21377,8 +_,10 @@
 			if (WorldGen.IsRope(num36, num38 - 1) || WorldGen.IsRope(num36, num38 + 1))
 				flag16 = true;
 
+			/*
 			if (Main.tile[num36, num37] == null)
 				Main.tile[num36, num37] = new Tile();
+			*/
 
 			if (!WorldGen.IsRope(num36, num37)) {
 				flag15 = false;
@@ -26280,8 +_,10 @@
 
 	public void TileInteractionsCheck(int myX, int myY)
 	{
+		/*
 		if (Main.tile[myX, myY] == null)
 			Main.tile[myX, myY] = new Tile();
+		*/
 
 		if (Main.tile[myX, myY].active()) {
 			TileInteractionsMouseOver(myX, myY);
@@ -29417,6 +_,8 @@
 
 	public void AdjTiles()
 	{
+		TilesWithinCraftingReach.Clear();
+
 		Array.Clear(adjTile, 0, adjTile.Length);
 		oldAdjWaterSource = adjWaterSource;
 		adjWaterSource = false;
@@ -29427,13 +_,85 @@
 		alchemyTable = false;
 		Rectangle tileRegion = TileReachCheckSettings.Simple.GetTileRegion(this, ateArtisanBread ? 4 : 0);
 		tileRegion = WorldUtils.ClampToWorld(tileRegion);
+
+		CraftingReachRectangle = tileRegion;
+
 		for (int i = tileRegion.Left; i <= tileRegion.Right; i++) {
 			for (int j = tileRegion.Top; j <= tileRegion.Bottom; j++) {
 				Tile tile = Main.tile[i, j];
 				if (tile.active()) {
 					SetAdjTile(tile.type);
+					/*
 					if (TileID.Sets.CountsAsWaterForCrafting[tile.type])
 						adjWaterSource = true;
+					*/
+
+					var isACraftingStation = Recipe.TileUsedInRecipes[tile.type];
+					if (TileID.Sets.CountsAsWaterForCrafting[tile.type]) {
+						adjWaterSource = true;
+						isACraftingStation = true;
+					}
+
+					if (isACraftingStation) {
+						TilesWithinCraftingReach.Add(new Point16(i, j));
+
+						if (i == tileRegion.Left) {
+							for (int x = tileRegion.Left - 1; x >= tileRegion.Left - 3; x--) {
+								if (x < 0) {
+									break;
+								}
+
+								if (tile.type == Main.tile[x, j].type) {
+									TilesWithinCraftingReach.Add(new Point16(x, j));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (i == tileRegion.Right) {
+							for (int x = tileRegion.Right + 1; x <= tileRegion.Right + 3; x++) {
+								if (x >= Main.maxTilesX) {
+									break;
+								}
+
+								if (tile.type == Main.tile[x, j].type) {
+									TilesWithinCraftingReach.Add(new Point16(x, j));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (j == tileRegion.Top) {
+							for (int y = tileRegion.Top - 1; y >= tileRegion.Top - 3; y--) {
+								if (y < 0) {
+									break;
+								}
+
+								if (tile.type == Main.tile[i, y].type) {
+									TilesWithinCraftingReach.Add(new Point16(i, y));
+								}
+								else {
+									break;
+								}
+							}
+						}
+						else if (j == tileRegion.Bottom) {
+							for (int y = tileRegion.Bottom + 1; y <= tileRegion.Bottom + 3; y++) {
+								if (y >= Main.maxTilesY) {
+									break;
+								}
+
+								if (tile.type == Main.tile[i, y].type) {
+									TilesWithinCraftingReach.Add(new Point16(i, y));
+								}
+								else {
+									break;
+								}
+							}
+						}
+					}
 				}
 
 				if (tile.liquid > 200 && tile.liquidType() == 0)
@@ -42100,8 +_,11 @@
 				if (whoAmI == Main.myPlayer && flag) {
 					int num7 = (int)(itemLocation.X + (float)heldItemFrame.Width * 0.8f * (float)direction) / 16;
 					int num8 = (int)(itemLocation.Y + mountOffset + (float)(heldItemFrame.Height / 2)) / 16;
+
+					/*
 					if (Main.tile[num7, num8] == null)
 						Main.tile[num7, num8] = new Tile();
+					*/
 
 					if (Main.tile[num7, num8].active() && TileID.Sets.Campfires[Main.tile[num7, num8].type] && Main.tile[num7, num8].frameY < 54) {
 						miscTimer++;
