--- src/TerrariaNetCore/Terraria/Projectile.cs
+++ src/Unified/Terraria/Projectile.cs
@@ -14075,7 +_,7 @@
 				return true;
 		}
 
-		if (StormLightningLiquidCollisionCheck(targetRect.Center()))
+		if (StormLightningLiquidCollisionCheck(targetRect.Center.ToVector2()))
 			return true;
 
 		if (customHitbox is MultiPointHitbox) {
@@ -45108,10 +_,12 @@
 		int num2 = 2;
 		int num3 = 0;
 		while (num3 <= num) {
+			int pY = p.Y;
 			if (flag)
-				FindBlockCeiling(p.X, ref p.Y, 10, 10);
+				FindBlockCeiling(p.X, ref pY, 10, 10);
 			else
-				FindBlockSurface(p.X, ref p.Y, 10, 10);
+				FindBlockSurface(p.X, ref pY, 10, 10);
+			p.Y = pY;
 
 			if (WorldGen.ActiveAndWalkableTile(p.X, p.Y)) {
 				Vector2 vector = p.ToWorldCoordinates(8f, 0f);
@@ -45432,8 +_,10 @@
 			Vector2 vector5 = default(Vector2);
 			for (int i = num10; i < num11; i++) {
 				for (int j = num12; j < num13; j++) {
+					/*
 					if (Main.tile[i, j] == null)
 						Main.tile[i, j] = new Tile();
+					*/
 
 					vector5.X = i * 16;
 					vector5.Y = j * 16;
@@ -45562,8 +_,11 @@
 		}
 		else if (ai[0] == 2f) {
 			Point point3 = base.Center.ToTileCoordinates();
+
+			/*
 			if (Main.tile[point3.X, point3.Y] == null)
 				Main.tile[point3.X, point3.Y] = new Tile();
+			*/
 
 			bool flag = true;
 			if (AI_007_GrapplingHooks_CanTileBeLatchedOnTo(Main.tile[point3.X, point3.Y]))
@@ -46138,11 +_,14 @@
 			int num4 = (int)(base.Center.Y / 16f);
 			_ = position.Y / 16f;
 			int num5 = (int)((position.Y + (float)height) / 16f);
+
+			/*
 			if (Main.tile[num3, num4] == null)
 				Main.tile[num3, num4] = new Tile();
 
 			if (Main.tile[num3, num5] == null)
 				Main.tile[num3, num5] = new Tile();
+			*/
 
 			if (velocity.Y > 0f)
 				velocity.Y *= 0.5f;
@@ -46349,11 +_,14 @@
 			int num8 = (int)(base.Center.Y / 16f);
 			_ = position.Y / 16f;
 			int num9 = (int)((position.Y + (float)height) / 16f);
+
+			/*
 			if (Main.tile[num7, num8] == null)
 				Main.tile[num7, num8] = new Tile();
 
 			if (Main.tile[num7, num9] == null)
 				Main.tile[num7, num9] = new Tile();
+			*/
 
 			if (velocity.Y > 0f)
 				velocity.Y *= 0.5f;
@@ -46431,6 +_,8 @@
 	private float AI_061_FishingBobber_GetWaterLine(int X, int Y)
 	{
 		float result = position.Y + (float)height;
+
+		/*
 		if (Main.tile[X, Y - 1] == null)
 			Main.tile[X, Y - 1] = new Tile();
 
@@ -46439,6 +_,7 @@
 
 		if (Main.tile[X, Y + 1] == null)
 			Main.tile[X, Y + 1] = new Tile();
+		*/
 
 		if (Main.tile[X, Y - 1].liquid > 0) {
 			result = Y * 16;
@@ -65112,8 +_,11 @@
 
 				int num726 = (int)position.X / 16;
 				int num727 = (int)position.Y / 16;
+
+				/*
 				if (Main.tile[num726, num727] == null)
 					Main.tile[num726, num727] = new Tile();
+				*/
 
 				if (Main.tile[num726, num727].type == 127 && Main.tile[num726, num727].active())
 					WorldGen.KillTile(num726, num727);
