--- src/TerrariaNetCore/Terraria/Rain.cs
+++ src/Unified/Terraria/Rain.cs
@@ -1,11 +_,12 @@
 using System;
 using Microsoft.Xna.Framework;
+using Terraria.Unified.Time;
 
 namespace Terraria;
 
 public class Rain
 {
-	public Vector2 position;
+	public DeltaVector2<DrawDelta> position;
 	public Vector2 velocity;
 	public float scale;
 	public float rotation;
@@ -25,7 +_,7 @@
 		if (Main.netMode == 2 || Main.gamePaused || Main.gameMenu || !Main.SceneMetrics.ZoneRain || Main.shimmerAlpha > 0f)
 			return;
 
-		float num = (float)Main.screenWidth / (float)Main.MaxWorldViewSize.X;
+		float num = (float)Main.screenWidth / (float)Main.MaxRenderingViewSize.X;
 		num *= 25f;
 		num *= 0.25f + 1f * Main.cloudAlpha;
 		if (NPC.AnyDanger(quickBossNPCCheck: true)) {
