--- src/TerrariaNetCore/Terraria/Initializers/AssetInitializer.cs
+++ src/Unified/Terraria/Initializers/AssetInitializer.cs
@@ -1,5 +_,7 @@
 using System;
+using System.Diagnostics;
 using System.IO;
+using System.Threading.Tasks;
 using Microsoft.Xna.Framework;
 using Microsoft.Xna.Framework.Graphics;
 using Newtonsoft.Json.Linq;
@@ -11,35 +_,61 @@
 using Terraria.GameContent;
 using Terraria.ID;
 using Terraria.IO;
-using Terraria.Testing;
 using Terraria.Utilities;
 
 namespace Terraria.Initializers;
 
 public static class AssetInitializer
 {
+	internal static AssetReaderCollection assetReaderCollection;
+
 	public static void CreateAssetServices(GameServiceContainer services)
 	{
+		XnbReader<Texture2D>.LoadOnMainThread = true;
+		XnbReader<DynamicSpriteFont>.LoadOnMainThread = true;
+		XnbReader<SpriteFont>.LoadOnMainThread = true;
+		XnbReader<Effect>.LoadOnMainThread = true;
+
+		assetReaderCollection = new AssetReaderCollection();
+		assetReaderCollection.RegisterReader(new PngReader(services.Get<IGraphicsDeviceService>().GraphicsDevice), ".png");
+		assetReaderCollection.RegisterReader(new XnbReader<Texture2D>(services), ".xnb");
+		assetReaderCollection.RegisterReader(new XnbReader<DynamicSpriteFont>(services), ".xnb");
+		assetReaderCollection.RegisterReader(new XnbReader<SpriteFont>(services), ".xnb");
+		assetReaderCollection.RegisterReader(new XnbReader<Effect>(services), ".xnb");
+		assetReaderCollection.RegisterReader(new FxcReader(Main.instance.Services.Get<IGraphicsDeviceService>().GraphicsDevice), ".fxc");
+
+		var assetPipeline = new AssetPipeline([], assetReaderCollection, 1, services);
+		var assetRepository = (IAssetRepository)assetPipeline;
+		assetRepository.AssetValueUpdatedHandler = (AssetValueUpdated)Delegate.Combine(assetRepository.AssetValueUpdatedHandler, new AssetValueUpdated(TagAsset));
+		services.AddService(typeof(AssetReaderCollection), assetReaderCollection);
+		services.AddService(typeof(IAssetRepository), assetRepository);
+
+		/*
 		AssetReaderCollection assetReaderCollection = new AssetReaderCollection();
 		assetReaderCollection.RegisterReader(new PngReader(services.Get<IGraphicsDeviceService>().GraphicsDevice), ".png");
 		assetReaderCollection.RegisterReader(new XnbReader(services), ".xnb");
 #if XNA && !SERVER
 		assetReaderCollection.RegisterReader(new FxReader(services.Get<IGraphicsDeviceService>().GraphicsDevice), ".fx");
 #endif
+
+		assetReaderCollection.RegisterReader(new FxcReader(Main.instance.Services.Get<IGraphicsDeviceService>().GraphicsDevice), ".fxc");
+
 		AsyncAssetLoader asyncAssetLoader = new AsyncAssetLoader(assetReaderCollection, 20);
 		asyncAssetLoader.RequireTypeCreationOnTransfer(typeof(Texture2D));
 		asyncAssetLoader.RequireTypeCreationOnTransfer(typeof(DynamicSpriteFont));
 		asyncAssetLoader.RequireTypeCreationOnTransfer(typeof(SpriteFont));
+
 		IAssetRepository assetRepository = new AssetRepository(new AssetLoader(assetReaderCollection), asyncAssetLoader);
 		assetRepository.AssetValueUpdatedHandler = (AssetValueUpdated)Delegate.Combine(assetRepository.AssetValueUpdatedHandler, new AssetValueUpdated(TagAsset));
 		services.AddService(typeof(AssetReaderCollection), assetReaderCollection);
 		services.AddService(typeof(IAssetRepository), assetRepository);
+		*/
 	}
 
-	private static void TagAsset(IAsset asset, object value)
+	private static void TagAsset(AssetRecord asset, object value)
 	{
 		if (value is GraphicsResource graphicsResource) {
-			graphicsResource.Name = asset.Name;
+			graphicsResource.Name = asset.Key.Path;
 			graphicsResource.Tag = asset;
 		}
 	}
