--- src/TerrariaNetCore/Terraria/GameContent/UI/Chat/RemadeChatMonitor.cs
+++ src/Unified/Terraria/GameContent/UI/Chat/RemadeChatMonitor.cs
@@ -35,6 +_,11 @@
 
 	public void AddNewMessage(string text, Color color, int widthLimitInPixels = -1)
 	{
+		// Safeguard against possible calls on the server.
+		if (Main.dedServ) {
+			return;
+		}
+
 		Trace.WriteLine("[chat] " + text);
 		if (!Main.dedServ) {
 			ChatMessageContainer chatMessageContainer = new ChatMessageContainer();
